### ISSUE TITLE
VBET Evidence / Likelihood of Valley Bottom Evidence Raster not displaying properly in ArcRAVE

### ISSUE THREAD

**Issue by shelbysawyer** [Riverscapes/riverscapes-tools#585]
When exploring the latest VBET runs in the staging server, I noticed that the VBET Evidence (Likelihood of Valley Bottom Raster) is displaying in an odd way. I don't know exactly how to describe it, other than to say that there are only a few scattered "clumps" of cells with data in them. I'll provide some pictures below to show what I mean. I will build this issue out more if our meeting reveals this is not just a misunderstanding on my part.

This is how it looks at the project (huc 8) level:
<img width="612" alt="evidence_huc8" src="https://user-images.githubusercontent.com/73311751/189374583-9734a7f0-61a2-4b7f-8c51-47e2e3ed8e3b.PNG">

This is how it looks zoomed in to one of the "clumps":
<img width="587" alt="evidence_zoom" src="https://user-images.githubusercontent.com/73311751/189374595-f3acf679-4d82-42e1-889a-8c6f22b9d620.PNG">

**Comment by joewheaton** [comment]
@KellyMWhitehead and @philipbaileynar? Ideas? I've had a play with this too and it seems like this isn't just a display problem. This seems like a write problem. This is the most fundamental output of VBET so lets make this a **top priority**. 

I'm guessing this has to do with reassembling the pieces properly (i.e. likelihoods for each of the levelpaths). From what I am seeing across projects, there are at least two problems: - You are not getting all the level path likelihood rasters reassembled (only some), OR in reassembling you are obliterating and writing over with not data the other pieces we want!
- You have a bug in writing data to disk in one of your loops as their are stripes of whole rows of data with no data in the middle of otherwise coherent(ish) data. 

## Questions:

1. Do you save to disk or just memory the intermediate likelihood rasters (by level path)?
2. Do we know that the intermediate rasters are working like we expect them to and producing correct looking data. We might need these to unpack why the fans are not working like we want.
3. How much disk space is used in the intermediate rasters if saved to disk?
4. When you write the intermediate (by levelpath) rasters, are you only writing it for the extent of the associated drainage wings, or for the whole raster? If so, do you clip the raster down to the extent or something else?
5. How do you reassemble (i.e. how do you know which rasters to use)?
6. Are you actually using this final raster to derive polygons, or the intermediates?

**Comment by KellyMWhitehead** [comment]
@joewheaton @philipbaileynar @shelbysawyer 

Yes, this was a write problem (windowing rasters with different extents). Today I was able to find a another method that fixes this bug.

For completeness, let me answer your questions:
1. The intermediate rasters are saved to a temp folder and wiped with each level path loop. I keep them around when debugging.
2. The intermediate rasters are working properly, so if we need to adjust our methods to improve the outputs (i.e. fans), that's where the changes will need to be made.
3. Disk space was/is a major concern as we generate a floating point raster (plus hand and a few for centerline) for each level path. Recently I made a change to limit the extents to a 400m square buffer of the channel area (to speed up processing), but there are still storage concerns retaining all these files (cybercastor and warehouse).
4. see above.
5. Raster reassembly is done as the last step of each level path loop using some clever numpy: if a cell is within a vbet polygon mask, the evidence value from that level path is used, otherwise the existing value (i.e. previous level paths) is used. Same for hand.
6. The composite raster is used for display only, the polygons are derived from the intermediates.

Happy to discuss or answer additional questions.
